src/views/citation_editable.tsx

```diff
@@ -110,20 +101,12 @@ export class CitationEditableView extends CitationView<
     }
 
     const handleSave = async (data: Partial<BibliographyItem>) => {
-      const ref = await saveModel({
-        ...data,
-      } as BibliographyItem)
-
-      const pos = findPosition(this.view.state.doc, ref._id)
-      if (pos) {
-        this.view.dispatch(
-          this.view.state.tr.setNodeMarkup(pos, undefined, {
-            id: ref._id,
-            containerTitle: ref['container-title'],
-            doi: ref.DOI,
-            ...ref,
-          })
-        )
+      if (data._id && findPosition(this.view.state.doc, data._id)) {
+        this.insertBibliographyNode(this.view, data as Build<BibliographyItem>)
```

Comment:
That is super super confusing. ok the goal is to just change prosemirror document and we don't need to touch modelMap at all, after we complete the task then will update modeMap

Reply:
I feel that we should always call saveModel/deleteModel even if it is not a new node; even if it's an update, its location needs to be determined based on other factors.

Alternatevaly, we should recreate _all_ item nodes when any reference change happens.

Reply:
Why won't we create a node if it doesn't exist?

Reply:
The problem here is the order, we don't know where should put this node, since its location is dictated by the citation style.

One more drastic approach is to:
- have all items (including ones that are not actually referenced) under the bibliography_element, but have them hidden.
- go back to rendering bibliography_element as an HTML blob. This would _not_ be stored in the models.
The benefits are:
- We can use citeproc the way it was designed, and call it once.
- We can track even nodes that are not referenced.
- We don't need to worry about item order. new nodes can be inserted at the end of the list.
The main problem is that we would need to do extra work to support the interactivity at the item level (TC, editing, comments).

Either way, references and citations work have been going on for months now, and I would like to have as much of it as possible finished before the next PI. Let's all have a call early next week to figure out what are the next steps here, or feel free to go on a call without me if it makes it easier to find a time.

Reply:
that is doable, for the work on the item level which will not require that much work as we can get the state of plugins for both comments and citations, and the work of LEAN-2723 will help to build tracking marks.

 if we are okay with Mohammad suggestion then there is nothing to do in this PR as in LEAN-2056 will implement that suggestion. @mbartenev-atypon @IslamJMomani is there anything I need to fix in this PR or do you think we need to set a call to sort it out

Reply:
> Why won't we create a node if it doesn't exist?

We don't want to create the node if the item is not cited. 



> One more drastic approach is to:
> 
> * have all items (including ones that are not actually referenced) under the bibliography_element, but have them hidden.
> * go back to rendering bibliography_element as an HTML blob. This would _not_ be stored in the models.

That sounds good to me. I have some concerns, but I will start working on refactoring the code and proceed with this approach. @mbartenev-atypon, do you have any concerns?



> @mbartenev-atypon @IslamJMomani is there anything I need to fix in this PR

For me, I think you are ready to go, as it's expected of me to handle the refactoring and manage what needs to be in the references section.